### PR TITLE
Added Space clone functionality

### PR DESCRIPTION
**Pull request by omanges**
Signed-off-by: Omkar Mestry <omkar.mestry@here.com>

### CODE REVIEW

File: xyzspaces/spaces.py [hunk 1]
```diff
@@ -1326,3 +1326,37 @@ def add_features_geopandas(
                 features_size=features_size,
                 chunk_size=chunk_size,
             )
+
+    def clone(self, space_id: str = None):
+        """
+        To copy current space data into a new space.
```
Comment (sackh) [inline xyzspaces/spaces.py:1332]: remove `To` just write `copy current space data into a new space`.

**Review by sackh — CHANGES_REQUESTED** [review]

**Review by sackh — APPROVED** [review]

**Review by deeplook — CHANGES_REQUESTED** [review]
Will comment later.

File: xyzspaces/spaces.py (hunk 1 above)
Comment (omanges) [reply xyzspaces/spaces.py]: Done changes

File: tests/space/test_space_objects.py [hunk 2]
```diff
@@ -791,3 +791,13 @@ def test_add_features_shapefile_diff_projection(empty_space):
         38.79930767,
         0,
     ]
+
+
+@pytest.mark.skipif(not XYZ_TOKEN, reason="No token found.")
+def test_space_clone(space_object, space_id, empty_space):
+    """Test for space clone functionality"""
+    space = space_object.read(id=space_id)
+    clone_space = space.clone()
```
Comment (deeplook) [inline tests/space/test_space_objects.py:800]: `cloned_space = ...`

File: tests/space/test_space_objects.py [hunk 3]
```diff
@@ -791,3 +791,13 @@ def test_add_features_shapefile_diff_projection(empty_space):
...
+    clone_specific_space = space.clone(space_id=empty_space.info["id"])
```
Comment (deeplook) [inline tests/space/test_space_objects.py:801]: `cloned_specific_space = ...`

File: tests/space/test_space_objects.py [hunk 4]
```diff
@@ -791,3 +791,13 @@ def test_add_features_shapefile_diff_projection(empty_space):
...
+    assert clone_space.get_statistics()["count"]["value"] == 180
```
Comment (deeplook) [inline tests/space/test_space_objects.py:802]: Would be nicer to make a deeper comparison than only for the number of features.

File: xyzspaces/spaces.py [hunk 5]
```diff
@@ -1326,3 +1326,39 @@ def add_features_geopandas(
                 features_size=features_size,
                 chunk_size=chunk_size,
             )
+
+    def clone(self, space_id: str = None, limit: int = 1000):
+        """
+        Copy current space data into a new space.
```
Comment (deeplook) [inline xyzspaces/spaces.py:1332]: Be more specific if the target space will be newly created or can be an existing one.

File: tests/space/test_space_objects.py (hunk 4 above)
Comment (omanges) [reply tests/space/test_space_objects.py:802]: This would slow down the test further, which we are trying to avoid

File: xyzspaces/spaces.py [hunk 6]
```diff
@@ -1326,3 +1326,39 @@ def add_features_geopandas(
...
+
+        :param space_id: space id into which to copy data,
+            if not provided will create a new space and copy the data.
+        :param limit: A max. number of features to read in a single iteration
```
Comment (deeplook) [inline xyzspaces/spaces.py:1336]: `limit` sounds misleading here, maybe better `chunks`?

File: xyzspaces/spaces.py [hunk 7]
```diff
@@ -1326,3 +1326,39 @@ def add_features_geopandas(
...
+            while iterating over the source space.
+        :return: The cloned Space Object
+        """
+        if space_id:
+            clone_space = Space.from_id(space_id=space_id)
```
Comment (deeplook) [inline xyzspaces/spaces.py:1341]: `cloned_space`

File: xyzspaces/spaces.py [hunk 8]
```diff
@@ -1326,3 +1326,39 @@ def add_features_geopandas(
...
+            features.append(f)
+            if len(features) == limit:
+                feature_collection = dict(
+                    type="FeatureCollection", features=features
+                )
+                clone_space.add_features(features=feature_collection)
+                features = []
+                feature_collection = {}
+
+        if len(features) != 0:
```
Comment (deeplook) [inline xyzspaces/spaces.py:1358]: Really `!= 0`, can it be negative? Better: `> 0`.

Comment (omanges) [reply xyzspaces/spaces.py:1358]: Iterate features can read >=0 features only

File: tests/space/test_space_objects.py [hunk 9]
```diff
@@ -791,3 +791,13 @@ def test_add_features_shapefile_diff_projection(empty_space):
         38.79930767,
         0,
     ]
+
+
+@pytest.mark.skipif(not XYZ_TOKEN, reason="No token found.")
+def test_space_clone(space_object, space_id, empty_space):
+    """Test for space clone functionality"""
```
Comment (deeplook) [inline tests/space/test_space_objects.py:798]: Test space cloning functionality.

File: xyzspaces/spaces.py (hunk 8 above)
Comment (deeplook) [reply xyzspaces/spaces.py:1358]: That's why.

File: tests/space/test_space_objects.py (hunk 4 above)
Comment (deeplook) [reply tests/space/test_space_objects.py:802]: You can also use a smaller test space for cloning.

**Review by omanges — COMMENTED** [review]
Done changes

File: xyzspaces/spaces.py [hunk 10]
```diff
@@ -1326,3 +1326,39 @@ def add_features_geopandas(
                 features_size=features_size,
                 chunk_size=chunk_size,
             )
+
+    def clone(self, space_id: str = None, chuks: int = 1000):
```
Comment (deeplook) [inline xyzspaces/spaces.py:1330]: Typo: chunks

File: xyzspaces/spaces.py [hunk 11]
```diff
@@ -1326,3 +1326,39 @@ def add_features_geopandas(
...
                 chunk_size=chunk_size,
             )
+
+    def clone(self, space_id: str = None, chunks: int = 1000):
+        """
+        Copy current space data into a newly created or into an existing space.
+
+        :param space_id: space id into which to copy data,
+            if not provided will create a new space and copy the data.
+        :param chuks: A max. number of features to read in a single iteration
```
Comment (deeplook) [inline xyzspaces/spaces.py:1336]: typo

**Review by deeplook — APPROVED** [review]

**Pull request merged**